examples/speaker_diarization/speaker_diarization/utils.py

```diff
@@ -204,10 +209,11 @@ def generate_identification_error(gt: GroundTruth, inf: Inference) -> List[TimeS
         gt_no = create_non_overlapping_segments(gt.transcription, id)
         inf_no = create_non_overlapping_segments(inf.transcription, id)
         res.extend(generate_fp(gt_no, inf_no))
+        res.extend(generate_fp(inf_no, gt_no))
```

Comment:
What are we trying to capture here by flipping the `inf` and `gt`?

Reply:
fp(gt, inf) calculates the FP, whereas fp(inf, gt) calculates the FN (which only works since we're working with intervals) - originally I had defined the error as just FP but it's more correct to include both FP and FN

Reply:
Got it. Do you think it makes sense to rename the function to be more relevant?

Reply:
Hmm, I think generate_identification_error is a descriptive enough function name as it is. Maybe we can create a generate_fn(gt, inf) function that just returns generate_fp(inf, gt), but that might be overkill? What are your thoughts?

Reply:
I think `generate_identification_error` is fine.  I just thought that `generate_fp` can be confusing when it can be used to extract fp and fn.  Why don't we call it `generate_error_segments` and leave a comment in its function header that explains two different usages?